### PR TITLE
Task-40910: Fix enable/disable the DLP feature

### PR DESCRIPTION
**Pull request by ahmadmido**
- Before this fix, only members of admins groups show the switch button on DLP feature.
-This fix will assure that all members on DLP group can show the switch button.

### CODE REVIEW

File: commons-component-common/src/main/java/org/exoplatform/settings/rest/ExoFeatureRest.java [hunk 1]
```diff
@@ -58,4 +58,34 @@ public Response isFeatureActiveForUser(
...
+          @ApiResponse(code = HTTPStatus.INTERNAL_ERROR, message = "Internal server error"), }
+  )
+  public Response changeFeatureActivation(
+      @ApiParam(value = "Feature name identifier", required = true) @PathParam("featureName") String featureName,
+      @ApiParam(value = "Is active feature", required = true) @PathParam("isActive") String isActive) {
+    if (StringUtils.isBlank(featureName)) {
+      return Response.status(Response.Status.BAD_REQUEST).entity("Feature name must not be null or blank").build();
+    }
+    try {
+      boolean isActiveBool = Boolean.parseBoolean(isActive);
```
Comment (rdenarie) [inline commons-component-common/src/main/java/org/exoplatform/settings/rest/ExoFeatureRest.java]: we should check taht the user is member of the dlp group.

**Review by rdenarie — CHANGES_REQUESTED** [review]

**Review by rdenarie — CHANGES_REQUESTED** [review]
2 things to change in this PR : 
The new rest path should be in DLP context, not in ExoFeatureRest : it should allow to activate/deactivate DLPFeature only to DLP admin only
In addition, the configuration of dlp group should be centralized in UserACL

File: commons-dlp/src/main/java/org/exoplatform/commons/dlp/rest/DlpRestServices.java [hunk 2]
```diff
@@ -0,0 +1,88 @@
...
+  private UserACL userACL;
+
+  public DlpRestServices(ExoFeatureService featureService, OrganizationService organizationService, UserACL userACL) {
+   this.featureService = featureService;
+   this.organizationService = organizationService;
+   this.userACL = userACL;
+  }
+
+
+  @Path("/changeFeatureActivation/{featureName}/{isActive}/")
```
Comment (rdenarie) [inline commons-dlp/src/main/java/org/exoplatform/commons/dlp/rest/DlpRestServices.java]: make parameter featureName static

File: commons-dlp/src/main/java/org/exoplatform/commons/dlp/rest/DlpRestServices.java [hunk 3]
```diff
@@ -0,0 +1,88 @@
...
+          @ApiResponse(code = HTTPStatus.INTERNAL_ERROR, message = "Internal server error"), }
+  )
+  public Response changeFeatureActivation(
+      @ApiParam(value = "Feature name identifier", required = true) @PathParam("featureName") String featureName,
+      @ApiParam(value = "Is active feature", required = true) @PathParam("isActive") String isActive) {
+    if (StringUtils.isBlank(featureName)) {
+      return Response.status(Response.Status.BAD_REQUEST).entity("Feature name must not be null or blank").build();
+    }
+    try {
+      if (!isMemberOfDlpGroup() && !userACL.isSuperUser() ) {
```
Comment (rdenarie) [inline commons-dlp/src/main/java/org/exoplatform/commons/dlp/rest/DlpRestServices.java]:
```suggestion
      if (!isMemberOfDlpGroup()) {
```

File: commons-dlp/src/main/java/org/exoplatform/commons/dlp/rest/DlpRestServices.java [hunk 4]
```diff
@@ -0,0 +1,88 @@
+package org.exoplatform.commons.dlp.rest;
+
+import io.swagger.annotations.*;
```
Comment (rdenarie) [inline commons-dlp/src/main/java/org/exoplatform/commons/dlp/rest/DlpRestServices.java]:
```suggestion
```

File: commons-dlp/src/main/java/org/exoplatform/commons/dlp/rest/DlpRestServices.java [hunk 5]
```diff
@@ -0,0 +1,88 @@
...
+import javax.ws.rs.PUT;
+import javax.ws.rs.Path;
+import javax.ws.rs.PathParam;
+import javax.ws.rs.Produces;
+import javax.ws.rs.core.MediaType;
+import javax.ws.rs.core.Response;
+import java.util.Collection;
+
+@Path("/features/dlp")
+@Api(value = "/features/dlp", description = "Manages Dlp features") // NOSONAR
```
Comment (rdenarie) [inline commons-dlp/src/main/java/org/exoplatform/commons/dlp/rest/DlpRestServices.java]:
```suggestion
@Api(value = "/features/dlp", description = "Manages Dlp features")
```

**Review by rdenarie — CHANGES_REQUESTED** [review]

File: commons-dlp/src/main/java/org/exoplatform/commons/dlp/rest/DlpRestServices.java (hunk 4 above)
Comment (ahmadmido) [reply commons-dlp/src/main/java/org/exoplatform/commons/dlp/rest/DlpRestServices.java]: It's necessary this import to use swagger annotations (@API, @ApiOperation ..)

File: commons-dlp/src/main/java/org/exoplatform/commons/dlp/rest/DlpRestServices.java [hunk 6]
```diff
@@ -0,0 +1,83 @@
...
+import javax.annotation.security.RolesAllowed;
+import javax.ws.rs.PUT;
+import javax.ws.rs.Path;
+import javax.ws.rs.PathParam;
+import javax.ws.rs.Produces;
+import javax.ws.rs.core.MediaType;
+import javax.ws.rs.core.Response;
+import java.util.Collection;
+
+@Path("/features/dlp")
```
Comment (azayati) [inline commons-dlp/src/main/java/org/exoplatform/commons/dlp/rest/DlpRestServices.java]: I think that /dlp is sufficient here, WDYT @rdenarie @ahmadmido ?

**Review by azayati — CHANGES_REQUESTED** [review]
See review

File: commons-dlp/src/main/java/org/exoplatform/commons/dlp/rest/DlpRestServices.java (hunk 6 above)
Comment (ahmadmido) [reply commons-dlp/src/main/java/org/exoplatform/commons/dlp/rest/DlpRestServices.java]: Ok (y)

Comment (rdenarie) [reply commons-dlp/src/main/java/org/exoplatform/commons/dlp/rest/DlpRestServices.java]: :+1:

**Review by rdenarie — APPROVED** [review]

**Review by azayati — APPROVED** [review]
LGTM

**Pull request merged**